src/components/LessonCard/LessonCard.tsx

```diff
@@ -108,61 +108,147 @@ const Paragraph = styled(P)`
   line-height: 19px;
 `;
 
+const Hr = styled.hr`
+  margin: var(--spacing-x-large) 0 var(--spacing-x-large) 0;
+`;
+
 const StyledButton = styled(Button)``;
 
+const selectMascot = (mascotType: MascotType, progress?: Progress) => {
+  let maskHead = false;
+  let maskBody = false;
+  let maskLegs = false;
+  if (progress) {
+    const progressRatio = progress.completed / progress.total;
+    maskHead = Boolean(progressRatio <= 2 / 3);
+    maskBody = Boolean(progressRatio <= 1 / 3);
+    maskLegs = Boolean(progressRatio <= 0 / 3);
+  }
+
+  let urlMascotSvg = "";
+  let urlMaskHeadSvg = "";
+  let urlMaskBodySvg = "";
+  let urlMaskLegsSvg = "";
+  switch (mascotType) {
+    case 1:
+      urlMascotSvg = urlLesson1MascotSvg;
+      urlMaskHeadSvg = urlLesson1MaskHeadSvg;
+      urlMaskBodySvg = urlLesson1MaskBodySvg;
+      urlMaskLegsSvg = urlLesson1MaskLegsSvg;
+      break;
+    case 2:
+      urlMascotSvg = urlLesson2MascotSvg;
+      urlMaskHeadSvg = urlLesson2MaskHeadSvg;
+      urlMaskBodySvg = urlLesson2MaskBodySvg;
+      urlMaskLegsSvg = urlLesson2MaskLegsSvg;
+      break;
+    case 3:
+      urlMascotSvg = urlLesson3MascotSvg;
+      urlMaskHeadSvg = urlLesson3MaskHeadSvg;
+      urlMaskBodySvg = urlLesson3MaskBodySvg;
+      urlMaskLegsSvg = urlLesson3MaskLegsSvg;
+      break;
+    case 4:
+      urlMascotSvg = urlLesson4MascotSvg;
+      urlMaskHeadSvg = urlLesson4MaskHeadSvg;
+      urlMaskBodySvg = urlLesson4MaskBodySvg;
+      urlMaskLegsSvg = urlLesson4MaskLegsSvg;
+      break;
+    case 5:
+    default:
+      urlMascotSvg = urlLesson5MascotSvg;
+      urlMaskHeadSvg = urlLesson5MaskHeadSvg;
+      urlMaskBodySvg = urlLesson5MaskBodySvg;
+      urlMaskLegsSvg = urlLesson5MaskLegsSvg;
+      break;
+  }
+
+  return (
+    <>
+      <img src={urlMascotSvg} />
+      {maskHead && <img src={urlMaskHeadSvg} />}
+      {maskBody && <img src={urlMaskBodySvg} />}
+      {maskLegs && <img src={urlMaskLegsSvg} />}
+    </>
+  );
+};
+
+type MascotType = 1 | 2 | 3 | 4 | 5;
+type Progress = {
+  complete: number;
+  total: number;
+};
+
 type Props = {
-  lessonId: string;
+  mascotType: MascotType;
+  progress?: Progress;
   subtitle?: string;
   title: string;
   description: string;
-  call2actionText?: string;
-  mascotType: MascotType;
+  buttonTextPrimary?: string;
+  buttonDestPrimary?: string;
+  buttonTextSecondaryDownload?: string;
```

Comment:
Can we not call this specific to "Download". Calling it secondary to be generic seems sufficient.

Reply:
Please see the answer below